server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -100,6 +100,8 @@ public class SearchExecutionContext extends QueryRewriteContext {
     private QueryBuilder aliasFilter;
     private boolean rewriteToNamedQueries = false;
 
+    private Integer requestSize;
```

Comment:
I see that this is required for when we transform the knnquerybuilder to a knnQuery. However, I don't think this should be nullable. The default & the provided values will not change during the lifetime of the request and are all known when this is created. 

So, instead of allowing this to be null, can we default it to `DEFAULT_SIZE`?

Reply:
Yeap, makes sense! Updated in 4082cf718f4999f9c1ee98a8b2a4ff04da6da845